Rest/Controller/UserController.php

```
@@ -120,6 +119,43 @@ public function deleteAction($id)
         return new Response("", 204);
     }
 
+
+    /**
+     * PATCH User
+     *
+     * @param int $id User ID
+     */
+    public function patchAction($id, Request $request)
+    {
+        if (!$this->isGranted('IS_AUTHENTICATED_FULLY')) {
+            throw new AccessDeniedException('You must be authenticated to view users');
+        }
+
+        $user = $this->getApplication()->getBBUserToken()->getUser();
+
+        if ($user->getId() !== $id) {
+            throw new AccessDeniedException('Only user can choose his password');
+        }
+
+        if ($user->getState() === User::PASSWORD_NOT_PICKED) {
+            $this->updateUserPassword($user, $request);
+        } else {
+            $encoderFactory = $this->getContainer()->get('security.context')->getEncoderFactory();
+            $oldPassword = $request->get('old_password', '');
```

Maybe you can throw an exception if ``$oldPassword`` is empty (``=== ''``) so you don't need to run the password encoding process. If i'm right you cannot create a password with an empty string or if it has not alteast 6 characters.